Formula/h/hdf5.rb

```diff
@@ -7,63 +7,66 @@ class Hdf5 < Formula
+    sha256 cellar: :any,                 sonoma:         "592edf06d0dd539e324787d4b658361e9e73554faea1100a16c595194774748a"
+    sha256 cellar: :any,                 ventura:        "dbea915eb8b1494edcca4f0df5bfdcb83d8467c86fb4de555da1779f28e8089a"
+    sha256 cellar: :any,                 monterey:       "ae5544130fdd0843e7b281d97a538aa307823f5dee429e681412e35d93ba5cd0"
+    sha256 cellar: :any_skip_relocation, x86_64_linux:   "2d3d1ccbf33cbbe10e13a6969d682217e650eefbaecc0f4a2e62c79394ce06a7"
   end
 
-  depends_on "cmake" => :build
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
   depends_on "gcc" # for gfortran
   depends_on "libaec"
-  depends_on "pkg-config"
+  depends_on "open-mpi"
 
   uses_from_macos "zlib"
 
-  conflicts_with "hdf5-mpi", because: "hdf5-mpi is a variant of hdf5, one can only use one or the other"
-
   def install
-    ENV["libaec_DIR"] = Formula["libaec"].opt_prefix.to_s
-    args = %w[
-      -DHDF5_USE_GNU_DIRS:BOOL=ON
-      -DHDF5_INSTALL_CMAKE_DIR=lib/cmake/hdf5
-      -DHDF5_BUILD_FORTRAN:BOOL=ON
-      -DHDF5_BUILD_CPP_LIB:BOOL=ON
-      -DHDF5_ENABLE_SZIP_SUPPORT:BOOL=ON
-    ]
-    system "cmake", "-S", ".", "-B", "build", *args, *std_cmake_args
+    # Work around incompatibility with new linker (FB13194355)
+    # https://github.com/HDFGroup/hdf5/issues/3571
+    ENV.append "LDFLAGS", "-Wl,-ld_classic" if DevelopmentTools.clang_build_version >= 1500
 
-    # Avoid c shims in settings files
-    inreplace_c_files = %w[
-      build/src/H5build_settings.c
-      build/src/libhdf5.settings
-      build/CMakeFiles/h5cc
-      build/CMakeFiles/h5hlcc
-    ]
-    inreplace inreplace_c_files, Superenv.shims_path/ENV.cc, ENV.cc
+    inreplace %w[c++/src/h5c++.in fortran/src/h5fc.in bin/h5cc.in],
+              "${libdir}/libhdf5.settings",
+              "#{pkgshare}/libhdf5.settings"
 
-    # Avoid cpp shims in settings files
-    inreplace_cxx_files = %w[
-      build/CMakeFiles/h5c++
-      build/CMakeFiles/h5hlc++
+    inreplace "src/Makefile.am",
+              "settingsdir=$(libdir)",
+              "settingsdir=#{pkgshare}"
+
+    if OS.mac?
+      system "autoreconf", "--force", "--install", "--verbose"
+    else
+      system "./autogen.sh"
+    end
+
+    args = %W[
+      --disable-dependency-tracking
+      --disable-silent-rules
+      --enable-build-mode=production
+      --enable-fortran
+      --enable-parallel
+      --prefix=#{prefix}
+      --with-szlib=#{Formula["libaec"].opt_prefix}
+      CC=mpicc
+      CXX=mpic++
+      FC=mpifort
+      F77=mpif77
+      F90=mpif90
     ]
-    inreplace_cxx_files << "build/src/libhdf5.settings" if OS.linux?
-    inreplace inreplace_cxx_files, Superenv.shims_path/ENV.cxx, ENV.cxx
+    args << "--with-zlib=#{Formula["zlib"].opt_prefix}" if OS.linux?
 
-    system "cmake", "--build", "build"
-    system "cmake", "--install", "build"
+    system "./configure", *args
+    system "make", "install"
```

Comment:
We don't want to get rid of CMake since upstream recommends CMake

Reply:
Done